### PR TITLE
rocAL - RPP & CuPy updates

### PR DESCRIPTION
**Pull request by kiritigowda**
* RPP - Name Updates
* rocAL - CuPy Install

### CODE REVIEW

**Comment by LakshmiKumar23** [comment]
@kiritigowda Tested on docker - ```compute-artifactory.amd.com:5000/rocm-plus-docker/compute-rocm-rel-5.5:13-ubuntu-20.04-stg1```

It fails to install cupy
```
python3
Python 3.8.10 (default, Nov 14 2022, 12:59:47) 
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import cupy
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'cupy'
```

Also, fails to import rocal pybind module
```
Traceback (most recent call last):
  File "rocAL_api_python_unittest.py", line 24, in <module>
    from amd.rocal.plugin.generic import ROCALClassificationIterator
ModuleNotFoundError: No module named 'amd'
```

**Comment by kiritigowda** [comment]
@LakshmiKumar23 the docker stg1 fails to install lib

```
sudo apt install -y hipblas hipsparse rocrand hipfft rocfft rocthrust-dev hipcub-dev 
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package hipblas
E: Unable to locate package hipsparse
E: Unable to locate package rocrand
E: Unable to locate package hipfft
E: Unable to locate package rocfft
E: Unable to locate package rocthrust-dev
E: Unable to locate package hipcub-dev
```

need to see why this is an issue with STG1 dockers, works on bare metal

**Comment by kiritigowda** [comment]
@LakshmiKumar23 pushed the changes, can you test now?

**Review by rrawther — APPROVED** [review]
I am not sure where the Cupy updates are for this PR?

**Comment by LakshmiKumar23** [comment]
@kiritigowda i am running the setup on stg1 docker again. I will approve once it passes

**Comment by LakshmiKumar23** [comment]
> I am not sure where the Cupy updates are for this PR?

@rrawther It's just in the setup script to install CuPy

File: MIVisionX-setup.py [hunk 1]
```diff
@@ -471,6 +472,19 @@
         os.system('sudo -v')
         os.system('(cd '+deps_dir+'; git clone -b '+rppVersion+' https://github.com/GPUOpen-ProfessionalCompute-Libraries/rpp.git; cd rpp; mkdir build-'+backend+'; cd build-'+backend+'; ' +
                   linuxCMake+' -DBACKEND='+backend+' -DCMAKE_INSTALL_PREFIX='+ROCM_PATH+' ../; make -j4; sudo make install)')
+        # CuPy Install
```
Comment (kiritigowda) [inline MIVisionX-setup.py:475]: @rrawther CuPy changes

**Comment by LakshmiKumar23** [comment]
@kiritigowda Fails to import
```
python3 
Python 3.8.10 (default, Nov 14 2022, 12:59:47) 
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import cupy
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/root/mivisionx-deps/cupy/cupy/__init__.py", line 32, in <module>
    import cupyx as _cupyx  # NOQA
  File "/root/mivisionx-deps/cupy/cupyx/__init__.py", line 8, in <module>
    from cupyx import linalg  # NOQA
  File "/root/mivisionx-deps/cupy/cupyx/linalg/__init__.py", line 2, in <module>
    from cupyx.linalg import sparse  # NOQA
  File "/root/mivisionx-deps/cupy/cupyx/linalg/sparse/__init__.py", line 3, in <module>
    from cupyx.linalg.sparse._solve import lschol  # NOQA
  File "/root/mivisionx-deps/cupy/cupyx/linalg/sparse/_solve.py", line 7, in <module>
    from cupyx.scipy import sparse
  File "/root/mivisionx-deps/cupy/cupyx/scipy/__init__.py", line 4, in <module>
    from cupyx.scipy.sparse._base import spmatrix as _spmatrix
  File "/root/mivisionx-deps/cupy/cupyx/scipy/sparse/__init__.py", line 1, in <module>
    from cupyx.scipy.sparse._base import issparse  # NOQA
  File "/root/mivisionx-deps/cupy/cupyx/scipy/sparse/_base.py", line 10, in <module>
    import scipy.sparse as _sparse
  File "/usr/lib/python3/dist-packages/scipy/sparse/__init__.py", line 229, in <module>
    from .base import *
  File "/usr/lib/python3/dist-packages/scipy/sparse/base.py", line 8, in <module>
    from .sputils import (isdense, isscalarlike, isintlike,
  File "/usr/lib/python3/dist-packages/scipy/sparse/sputils.py", line 16, in <module>
    supported_dtypes = [np.typeDict[x] for x in supported_dtypes]
  File "/usr/lib/python3/dist-packages/scipy/sparse/sputils.py", line 16, in <listcomp>
    supported_dtypes = [np.typeDict[x] for x in supported_dtypes]
  File "/usr/local/lib/python3.8/dist-packages/numpy/__init__.py", line 320, in __getattr__
    raise AttributeError("module {!r} has no attribute "
AttributeError: module 'numpy' has no attribute 'typeDict'
```

**Comment by kiritigowda** [comment]
@LakshmiKumar23 looks like it needs a very particular version of numpy - `pip install numpy==1.21`  with this there is no issue. See below. Try this out and let me know.

```
root@kapu:/src# python3 
Python 3.8.10 (default, Nov 14 2022, 12:59:47) 
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import cupy
>>>
```

**Comment by LakshmiKumar23** [comment]
> @LakshmiKumar23 looks like it needs a very particular version of numpy - `pip install numpy==1.21` with this there is no issue. See below. Try this out and let me know.
> 
> ```
> root@kapu:/src# python3 
> Python 3.8.10 (default, Nov 14 2022, 12:59:47) 
> [GCC 9.4.0] on linux
> Type "help", "copyright", "credits" or "license" for more information.
> >>> import cupy
> >>>
> ```

@kiritigowda Yes, CuPy installs now. Still seeing an issue with rocal_pybind. Can open a separate issue for that. This PR can be merged

**Pull request merged**